docs/e2e.md

```diff
@@ -0,0 +1,217 @@
+- [`npm-check`](./tools/src/npm/README.md#npm-check) - check if the package is installed in registry
+- [`npm-install`](./tools/src/npm/README.md#npm-install) - install package in workspace.
+- [`npm-uninstall`](./tools/src/npm/README.md#npm-uninstall) - uninstall package form workspace
+
+The following steps are necessary to publish a package:
+
+1. `nx run <project-name>:npm-check` - check if the package is not already published
+2. `nx run <project-name>:publish --nextVersion=<version>` - publish package (login required)
+3. `nx run <project-name>:npm-check` - check if the package is published
+
+## E2E testing
+
+> [!NOTE] Projects that need verdaccio are identified over the `e2e` target.
+> Those libraries will have dynamic targets to start verdaccio and test the package.
+
+All E2E tests use verdaccio to test the build artefact in a real registry.
+
+Every E2E project in the monorepo has the following targets:
+
+- [`start-verdaccio`](./tools/src/verdaccio/README.md#start-verdaccio) - start a local registry
+
+#### Running E2E tests for a given project:
+
+Every project in the monorepo that has E2E tests follows the project naming pattern: `<project-name>-e2e`.
+
+Examples:
+
+- `npx nx e2e cli-e2e` - run E2E tests for the cli project
+- `npx nx e2e cli-e2e --skipNxCache` - pass Nx CLI arguments
+- `npx nx run-many -t e2e` - run all E2E tests
+
+### E2E testing process
+
+The `e2e` testing process is complex and involves multiple steps and targets.
+
+#### Overview:
+
+- `nx run e2e <project-name>`
+  - `global-setup.e2e.ts#setup` (vitest setup script configured in `vite.config.e2e.ts`)
+    - setup - `nx start-verdaccio`
+    - setup - `nx run-many -t publish`
+    - setup - `nx run-many -t npm-install`
+  - **run tests**
+  - `global-setup.e2e.ts#teardown` (vitest teardown script configured in `vite.config.e2e.ts`)
+    - teardown - `nx run-many -t npm-uninstall`
+    - teardown - `process.kill(<verdaccio-port>)`
+
+```mermaid
+graph TD
+  A["nx run e2e <project-name>"] --> B[global-setup.e2e.ts]
+  B --> C[nx start-verdaccio]
+  C --> D[nx run-many -t publish]
+  D --> E[nx run-many -t npm-install]
+  E --> F[vitest test]
+  F --> G[nx run-many -t npm-uninstall]
+  G --> H["process.kill(<verdaccio-port>)"]
+```
+
+#### Involved files:
+
+```text
+Root/
+├── e2e/
+│   └── <project-name>-e2e/
+│       ├── tests/
+│       │   └── <file-name>.e2e.ts
+│       ├── vite.config.e2e.ts // uses `global-setup.e2e.ts` for as `globalSetup` script
+│       └── project.json
+├── packages/
+│   └── <project-name>/
+│       └── project.json // marked as "publishable"
+├── .verdaccio/
+│   └── config.yaml
+├── tools/ // all plugins registered in nx.json
+│    └── src/
+│       ├── npm/
+│       │   └── npm.plugin.ts
+│       ├── publish/
+│       │    └── publish.plugin.ts
+│       └── verdaccio/
+│           └── verdaccio.plugin.ts
+├── global-setup.e2e.ts
+└── nx.json // registers npm, publish and verdaccio plugin
```

Comment:
A little "hack" to get some useful syntax highlighting for comments and variables:

<details><summary> :framed_picture: before </summary>
<p>

![image](https://github.com/user-attachments/assets/3ade686c-807b-432a-bd1f-89d5c92cbad8)


</p>
</details> 

<details><summary> :framed_picture: after</summary>
<p>

![image](https://github.com/user-attachments/assets/a6ec103f-77ed-4ec3-b0ef-67cf53400a99)


</p>
</details> 

```suggestion
```sh
Root/
├── e2e/
│   └── <project-name>-e2e/
│       ├── tests/
│       │   └── <file-name>.e2e.ts
│       ├── vite.config.e2e.ts # uses `global-setup.e2e.ts` for as `globalSetup` script
│       └── project.json
├── packages/
│   └── <project-name>/
│       └── project.json # marked as "publishable"
├── .verdaccio/
│   └── config.yaml
├── tools/ # all plugins registered in nx.json
│    └── src/
│       ├── npm/
│       │   └── npm.plugin.ts
│       ├── publish/
│       │    └── publish.plugin.ts
│       └── verdaccio/
│           └── verdaccio.plugin.ts
├── global-setup.e2e.ts
└── nx.json # registers npm, publish and verdaccio plugin
```

(I thought I'd already commented on this somewhere, but can't find it :sweat_smile:)

Reply:
I remember the `json` -> `jsonc`